### PR TITLE
EL-1229: Amalgamate dependabot PRs

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1229)

## What changed and why

Adapt a public workflow to combine PRs
(What I changed was to add automatic scheduling of the workflow to run on a Monday morning. Because of that, I needed the script to use default values for variables that would be specified when the workflow was run manually, so I added a few `|| 'some-string'` to a few places.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: .github/workflows/combine-prs.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                mergeFailedPRs.push(prString);
+              }
+            }
+
+            console.log('Creating combined PR');
+            const combinedPRsString = combinedPRs.join('\n');
+            let body = '✅ This PR was created by the Combine PRs workflow by combining the following PRs:\n' + combinedPRsString;
+            if(mergeFailedPRs.length > 0) {
+              const mergeFailedPRsString = mergeFailedPRs.join('\n');
+              body += '\n\n⚠️ The following PRs were left out due to merge conflicts:\n' + mergeFailedPRsString
```
Comment (MazOneTwoOne) [inline .github/workflows/combine-prs.yml:141]: Is that an emoji/wingding?

File: .github/workflows/combine-prs.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+# Controls when the action will run - in this case triggered manually
+on:
+  workflow_dispatch:
+    inputs:
+      branchPrefix:
+        description: 'Branch prefix to find combinable PRs based on'
+        required: true
+        default: 'dependabot'
+      combineBranchName:
```
Comment (MazOneTwoOne) [inline .github/workflows/combine-prs.yml:12]: question - so we want to have a PR which has every dependabot update, regardless of whether is passing tests (I see we have omitted `mustBeGreen`)? I guess we will still have the existing dependabot PRs so we can see where to focus our search for the combined PR???

**Review by MazOneTwoOne — APPROVED** [review]
It's copied over as suggested in the other resource and can see where and why you've made changes.

File: .github/workflows/combine-prs.yml (hunk 2 above)
Comment (patrick-laa) [reply .github/workflows/combine-prs.yml:12]: We don't have mustBeGreen as a variable, but what we've removed from the code is the path where mustBeGreen was set to false. Now it's hard-coded that the branch must be green to be included in the amalgamation, there's no way of bypassing it. The reason being that it was really hard to have a boolean variable that defaulted to true when the workflow was called by the scheduler (in which case this `inputs` section is ignored), and it seemed to me unlikely we'd ever want to combine failing PRs.

File: .github/workflows/combine-prs.yml (hunk 1 above)
Comment (patrick-laa) [reply .github/workflows/combine-prs.yml:141]: It's an emoji that seems only to render as a colourful emoji in some contexts.

**Pull request merged**